api/v1beta2/foundationdbcluster_types.go

```diff
@@ -218,6 +219,11 @@ type FoundationDBClusterSpec struct {
 	// UseUnifiedImage determines if we should use the unified image rather than
 	// separate images for the main container and the sidecar container.
 	UseUnifiedImage *bool `json:"useUnifiedImage,omitempty"`
+
+	// MaxZonesWithUnavailablePods defines the maximum number or percent of zones that can have unavailable pods during the update process.
+	// When set to 0 there is no limit on the number or percent of zones with unavailable pods.
+	// +kubebuilder:default:=0
+	MaxZonesWithUnavailablePods intstr.IntOrString `json:"maxZonesWithUnavailablePods,omitempty"`
```

Comment:
I think in this case the percentage doesn't make sense as the zones can be quite dynamic, so converting this to an `*int` would be better.